lfs/gitfilter_smudge.go

```diff
@@ -86,22 +86,50 @@ func (f *GitFilter) Smudge(writer io.Writer, ptr *Pointer, workingfile string, d
 func (f *GitFilter) downloadFile(writer io.Writer, ptr *Pointer, workingfile, mediafile string, manifest *tq.Manifest, cb tools.CopyCallback) (int64, error) {
 	fmt.Fprintf(os.Stderr, "Downloading %s (%s)\n", workingfile, humanize.FormatBytes(uint64(ptr.Size)))
 
-	// NOTE: if given, "cb" is a tools.CopyCallback which writes updates
-	// to the logpath specified by GIT_LFS_PROGRESS.
-	//
-	// Either way, forward it into the *tq.TransferQueue so that updates are
-	// sent over correctly.
-
-	q := tq.NewTransferQueue(tq.Download, manifest, f.cfg.Remote(),
-		tq.WithProgressCallback(cb),
-		tq.RemoteRef(f.RemoteRef()),
-	)
-	q.Add(filepath.Base(workingfile), mediafile, ptr.Oid, ptr.Size)
-	q.Wait()
-
-	if errs := q.Errors(); len(errs) > 0 {
+	mainRemote := f.cfg.Remote()
+	remotes := append([]string{mainRemote}, f.cfg.Remotes()...)
+	for i, remote := range remotes {
+		if remote == mainRemote {
```

Comment:
Since you initialize `remotes` as:

```go
removes := append([]string{mainRemote}, f.cfg.Remotes()...)
```

I think that it would always be the case that `remotes[0] == mainRemote`, and hence this loop would terminate in the same fashion after only a single iteration. Can you clarify its original intent, and see whether or not it is necessary?

Reply:
Ah nice catch. My intent was to have a list of all remotes but keep the remote suggested by the original heuristics in front, so behaviour would only change if an object is missing.

Basically, `remotes` should be initialised as `f.cfg.Remotes()`.